website_product_comment_purchased/models/mail.py

```diff
@@ -22,41 +22,41 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from openerp import models, fields, api
+from openerp.osv import osv, fields
 
 
-class MailMessage(models.Model):
+class MailMessage(osv.Model):
     _inherit = 'mail.message'
 
-    @api.multi
-    def _comment_bought(self):
-        sale_report_obj = self.env['sale.report']
-        sale_select = sale_report_obj._select()
-        sale_from = sale_report_obj._from()
-        sale_group_by = sale_report_obj._group_by()
-        for mail in self:
-            where = """
+    def _comment_bought(self, cr, uid, ids, name, arg, context=None):
+        res = {}.fromkeys(ids, False)
+        sale_report_obj = self.pool.get('sale.report')
+        mail_cache = self.browse(cr, uid, ids, context)[0]
```

Comment:
Use a for to set value in all ids and get the boolean by each id.
Similar to new api script